test/franklin_web/admin-user-stories/can_create_and_edit_with_article_editor_test.exs

```diff
@@ -0,0 +1,189 @@
+
+    # This will be the article we will edit.
+    edit_article_attributes = %{
+      title: "edit article title",
+      body: "edit article body",
+      published_at: now
+    }
+
+    {:ok, edit_article_id} = Articles.create_article(edit_article_attributes)
+
+    edit_article =
+      wait_for_passing(fn ->
+        # Because projections are not instant, we need to wait until it is finished.
+        assert %Article{
+                 title: "edit article title",
+                 body: "edit article body",
+                 published_at: ^now
+               } = Articles.get_article(edit_article_id)
+      end)
+
+    {:ok, edit_view, _html} = live(conn, "/admin/articles/editor/#{edit_article.id}")
+
+    ~M{create_view, edit_view, edit_article}
+  end
+
+  test "creation succeeds with all required form fields", ~M{create_view} do
+    valid_params = %{
+      title: "A valid new article title.",
+      body: "A valid new article body."
+    }
+
+    create_view
+    |> form("#new-article", article_form: valid_params)
+    |> render_submit()
+
+    assert {"/admin/articles", _flash} = assert_redirect(create_view)
+
+    # Because the data projection can take time, we need to wait_for_passing.
+    wait_for_passing(fn ->
+      # FIXME: This is a shit test because I'm using `list_articles/0` but atm I don't
+      # have the `id` of the created article to use in `get_article/1`. Once we
+      # change to redirecting to a view page we can sniff the id from the url
+      # being redirected to.
+      assert Enum.find(Articles.list_articles(), nil, fn article ->
+               match?(
+                 %Article{
+                   title: "A valid new article title.",
+                   body: "A valid new article body."
+                 },
+                 article
+               )
+             end)
+    end)
+  end
+
+  test "editing succeeds with all required form fields changed", ~M{edit_view, edit_article} do
+    edited_title = "#{edit_article.title} was edited."
+    edited_body = "#{edit_article.body} was edited."
+    edited_published_at = DateTime.add(edit_article.published_at, -1, :day)
+
+    edited_params = %{
+      title: edited_title,
+      body: edited_body,
+      published_at: DateTime.to_iso8601(edited_published_at)
+    }
+
+    edit_view
+    |> form("#new-article", article_form: edited_params)
+    |> render_submit()
+
+    # Because the data projection can take time, we need to wait_for_passing.
+    wait_for_passing(fn ->
+      assert %Article{
+               title: ^edited_title,
+               body: ^edited_body,
+               published_at: ^edited_published_at
+             } = Articles.get_article(edit_article.id)
+    end)
+
+    assert {"/admin/articles", _flash} = assert_redirect(edit_view)
+  end
+
+  describe "verify title input failure responses" do
+    test "fails without a required title value", ~M{create_view, edit_view} do
+      for view <- [create_view, edit_view] do
```

Comment:
Doing the same test for each view type can feel a bit noisy and redundant here. That said, I thought this was a better solution than making a separate explicit story for both creation and editing -- I'd much rather just have a user story for the "editor" since it does both things. Feedback welcome. 🤷‍♂️